confirmation_wizard/wizard/confirmation_message_wizard.py

```diff
@@ -0,0 +1,84 @@
+# Copyright (C) 2024 Cetmix OÜ
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, fields, models
+
+
+class ConfirmationMessageWizard(models.TransientModel):
+    _name = "confirmation.message.wizard"
+    _description = "Confirmation Wizard"
+
+    text = fields.Char(string="Confirm Message", required=True)
+
+    res_ids = fields.Char()
+    res_model = fields.Char()
+
+    callback_method = fields.Char()
+    callback_params = fields.Json()
+
+    return_type = fields.Selection(
+        [
+            ("window_close", "Return Window Close Action"),
+            ("method", "Return Method"),
+        ],
+        default="window_close",
+    )
+
+    @api.model
+    def _prepare_action(self, title=None):
+        action = self.env["ir.actions.actions"]._for_xml_id(
+            "confirmation_wizard.confirmation_message_wizard_action"
+        )
+        if title:
+            action["name"] = title
+        action["context"] = self._context
+        return action
+
+    @api.model
+    def confirm_message(
+        self, text, records, title=None, method=None, callback_params=None
+    ):
+        callback_params = callback_params or {}
+        res_ids = ",".join(map(str, records.ids))
+        wizard = self.create(
+            {
+                "text": text,
+                "res_ids": res_ids,
+                "return_type": "method",
+                "res_model": records._name,
+                "callback_method": method,
+                "callback_params": callback_params,
+            }
+        )
+        return wizard._prepare_action(title)
+
+    @api.model
+    def confirm_no_action_message(self, text, title=None):
+        wizard = self.create(
+            {
+                "text": text,
+                "return_type": "window_close",
+            }
+        )
+        return wizard._prepare_action(title)
```

Comment:
The methods confirm_message and confirm_no_action_message could be merged by passing the return type as a parameter, which would reduce code duplication. e.g:

```python
    @api.model
    def confirm_message(
        self, message, records=None, title=None, method=None, callback_params=None, return_type="method"
    ):
        """Creates and returns the confirmation wizard action."""
        callback_params = callback_params or {}
        wizard = self.create({
            "message": message,
            "record_ids": ",".join(map(str, records.ids)) if records else "",
            "res_model": records._name if records else "",
            "callback_method": method,
            "callback_params": callback_params,
            "return_type": return_type,
        })
        return wizard._prepare_action(title)
```